status/private/waku.nim

```diff
@@ -2,28 +2,352 @@
 # nim-confutils. The compiler will force nim-confutils to annotate its procs
 # with the needed `{.raises: [,,,].}` pragmas.
 
+{.push raises: [Defect].}
+
+# ------------------------------------------------------------------------------
+
 import # std libs
-  std/[options, os]
+  std/[json, options, random, sequtils, strutils, tables, times, uri]
 
 import # vendor libs
-  confutils, chronicles, chronos,
-  libp2p/crypto/[crypto, secp],
-  eth/keys,
-  json_rpc/[rpcclient, rpcserver],
-  stew/shims/net as stewNet,
-  waku/v2/node/[config, wakunode2],
-  waku/common/utils/nat
+  bearssl, chronicles, chronos, chronos/apps/http/httpclient, eth/keys,
+  libp2p/[crypto/crypto, crypto/secp, multiaddress, muxers/muxer, peerid,
+          peerinfo, protocols/protocol, stream/connection, switch],
+  nimcrypto/utils,
+  stew/[byteutils, endians2, results, shims/net],
+  waku/common/utils/nat,
+  waku/v2/node/[waku_payload, wakunode2],
+  waku/v2/protocol/[waku_filter/waku_filter, waku_lightpush/waku_lightpush,
+                    waku_message, waku_store/waku_store],
+  waku/v2/utils/peers
+
+import # status modules
+  ./protocol as statusProtocol
+
+export byteutils, crypto, keys, nat, net, peers, results, secp, utils,
+       waku_filter, waku_lightpush, waku_message, waku_store, wakunode2
+
+logScope:
+  topics = "client"
+
+type
+  PrivateKey* = crypto.PrivateKey
 
-# The initial implementation of initNode is by intention a minimum viable usage
-# of nim-waku v2 from within nim-status
+  Topic* = wakunode2.Topic
 
-proc initNode*(config: WakuNodeConf = WakuNodeConf.load()): WakuNode {.raises:
-  [Defect, Exception].} =
+const DefaultTopic* = "/waku/2/default-waku/proto"
 
+proc selectRandomNode*(fleetStr: string): Future[string] {.async.} =
   let
-    (extIp, extTcpPort, extUdpPort) = setupNat(config.nat, clientId,
-      Port(uint16(config.tcpPort) + config.portsShift),
-      Port(uint16(config.udpPort) + config.portsShift))
+    url = "https://fleets.status.im"
```

Comment:
Maybe we should `strdefine` this or pull out to a constant somewhere. I know we need to do this with the OpenSea and cryptocompare API URLs as well, so maybe could handle it later.

Reply:
Some good news for this one is that DNS based discovery is being worked on by the vac/waku team, which will replace the need to bootstrap by downloading JSON from https://fleets.status.im.